frontend/src/truck/useTrucks.ts

```diff
@@ -22,7 +28,28 @@ export default function useTrucks() {
             name,
         }
         return axios.post("/api/trucks", newTruck)
-            .then(() => getAllTrucks())
+            .then(() => fetchAllTrucks())
     }
-    return {trucks, addTruck}
+
+    const deleteTruck = (id: string) => {
+        return axios.delete("/api/trucks/" + id)
+            .then((response) => response.status)
+            .then(fetchAllTrucks)
+            .catch(error => notify("existiert nicht"));
+    }
+
+    const getTruckById = (id: string | undefined) => {
+        return trucks.find(thisTruck => {
+            if (thisTruck.id === id) return true
```

Comment:
Tipp: könnte man noch kürzer formulieren:
```suggestion
            return thisTruck.id === id
```